Tests/test_Seq_objs.py

```diff
@@ -737,6 +737,165 @@ def test_init_typeerror(self):
         self.assertRaises(TypeError, Seq, (1066))
         self.assertRaises(TypeError, Seq, (Seq("ACGT", generic_dna)))
 
+    def test_join_Seq(self):
+        """Checks if Seq join correctly concatinates sequence with the spacer."""
+        # Only expect it to take Seq objects and/or strings in an iterable!
+
+        filename = 'Fasta/f003'
+        seqlist = SeqIO.parse(filename, 'fasta')
+        spacer = Seq('NNNNN')
+        spacer1 = Seq('')
+        spacer2 = Seq('GGG', generic_dna)
+        example_strings = ["ATG", "ATG", "ATG", "ATG"]
+        test_seq = Seq("ATG", generic_dna)
+        example_strings_seqs = ["ATG", "ATG", test_seq, "ATG"]
+
+        # seq objects with spacer
+        concatenated = spacer.join(record.seq for record in SeqIO.parse(filename, 'fasta'))
+        # seq objects with empty spacer
+        concatenated1 = spacer1.join(record.seq for record in SeqIO.parse(filename, 'fasta'))
+        # strings with empty spacer
+        concatenated2 = spacer1.join(example_strings)
+        # strings and seqs
+        concatenated3 = spacer.join(example_strings_seqs)
+        concatenated4 = spacer1.join(example_strings_seqs)
+        concatenated5 = spacer2.join(example_strings_seqs)
+
+        temp_data = temp_data1 = temp_data2 = temp_data3 = ""
```

Comment:
You have named things ``concatenatedN`` and ``temp_dataN`` but they are out of sync.

Using ``str_concatenatedN`` and ``seq_concatenatedN`` would be much clearer, where we expect ``str_concatenatedN == str(seq_concatenatedN)`` for each ``N``.